### PR TITLE
DSS-127 Allow/test record/file copy with records > 2MB of data

### PR DESCRIPTION
**Pull request by perrymanmd**

### CODE REVIEW

**Review by ktarbet — CHANGES_REQUESTED** [review]
It doesn't seem proper to see the submodule changes in the PR.

File: heclib/heclib_c/src/Internal/zCopyRecord6Dynamic.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#include <heclib.h>
+#include <string.h>
+
+#define KIHEAD 1000
+#define KCHEAD 100
+
+#ifdef _MSC_VER
+char* strndup(const char* src, size_t max_len) {
```
Comment (ktarbet) [inline heclib/heclib_c/src/Internal/zCopyRecord6Dynamic.c:8]: could we use the method stringFortToC instead?     

note: There is a bug in stringFortToC  where it doesn't trim more than one trailing space.

Comment (perrymanmd) [reply heclib/heclib_c/src/Internal/zCopyRecord6Dynamic.c:8]: We could, but shouldn't we fix the bug first? I usually trust the C library functions when they exist - it's just that for some reason Microsoft doesn't include this function anymore.

Comment (perrymanmd) [reply heclib/heclib_c/src/Internal/zCopyRecord6Dynamic.c]: Didn't find any bug in `stringFortToC()`. It only nulls one character, but it only allocates one more than the number of non-trailing-space characters of the Fortran string, so one is enough.

**Pull request merged**